appinventor/appengine/src/com/google/appinventor/client/explorer/youngandroid/ProjectToolbar.java

```diff
@@ -157,7 +157,8 @@ public void onSuccess(Void result) {
               Ode.getInstance().getProjectManager().removeProject(projectId);
               // Show a welcome dialog in case there are no
               // projects saved.
-              if (Ode.getInstance().getProjectManager().getProjects().size() == 0) {
+              if (Ode.getInstance().getProjectManager().getProjects().size() == 0
+                  && Ode.getInstance().getCurrentView() != 2) { //2 is the int for Gallery
```

Comment:
What is the 'int for the Gallery'? Can this be a **constant with a proper name** instead of a **magic number**?